src/classes/THAN_Filter_TDTM.cls

```diff
@@ -61,12 +61,47 @@ public class THAN_Filter_TDTM extends TDTM_Runnable {
 	                    String filterField = String.escapeSingleQuotes(handler.Filter_Field__c);
 	                    String obj = String.escapeSingleQuotes(handler.Object__c);
 	                    if(!String.isBlank(filterField) && !String.isBlank(obj)) {
-		                    try {
-		                        String dynamicQuery = 'select ' + filterField + ' from ' + obj + ' limit 1';
-		                        List<SObject> records =  Database.query(dynamicQuery);
-		                    } catch(QueryException e) {
-		                        handler.addError(Label.InvalidFilter + handler.Class__c);
-		                    }
+                            //those four type of fields are considered valid Filter field
+                            //1. Standard Field e.g. FirstName
+                            //2. HEDA Field WITHOUT namespace e.g. Military_Service__c
+                            //3. User's Custom Field e.g. MyField__c
+                            //4. Field from another package with namespace e.g. NS__Field__c
+
+                            String hedaPrefix = String.isBlank(UTIL_Namespace.getNamespace()) ? '' : (UTIL_Namespace.getNamespace() + '__');
+                            Boolean hasHEDAPrefix = filterField.startsWithIgnoreCase(hedaPrefix) ? true : false;
+                            Boolean ignoreDuplicateField = filterField.endsWithIgnoreCase(Label.duplicateFieldInFilter) ? true: false;
+
+                            //Get the ignoreDuplicateField flag and remove it from the Filter Field value
+                            if(ignoreDuplicateField) {
+                                filterField = filterField.replace(Label.duplicateFieldInFilter, '');
+                                handler.Filter_Field__c = handler.Filter_Field__c.replace(Label.duplicateFieldInFilter, '');
+                            }
+
+                            //System should't allow a Filter Field with HEDA namespace
+                            if(hasHEDAPrefix) {
+                                handler.addError(Label.InvalidFilter + handler.Class__c + '. ' + Label.fieldWithHEDANamespace);
+                            } else {
+                                Boolean fieldExists = false;
+                                Boolean fieldExistsHEDA = false;
```

Comment:
Not a big deal, but these two variables could be set the first time they are used, without initializing them first (we could just remove lines 84 and 85).

Reply:
right